### PR TITLE
Fleshing out the SqlProjects service bindings

### PR DESCRIPTION
**Pull request by Benjin**
Adding all remaining bindings for the new SqlProjects service that'll handle all XML manipulation operations for the SQL Database Projects extension

### CODE REVIEW

File: extensions/mssql/src/mssql.d.ts [hunk 1]
```diff
@@ -319,16 +319,95 @@ declare module 'mssql' {
 	//#region --- Sql Projects
 
 	export interface ISqlProjectsService {
+		addDacpacReference(projectUri: string, dacpacPath: string, suppressMissingDependencies: boolean, databaseVariable?: string, serverVariable?: string, databaseLiteral?: string): Promise<azdata.ResultStatus>;
```
Comment (Charles-Gagnon) [inline extensions/mssql/src/mssql.d.ts:322]: Doc comments for new things in this file would be helpful.

File: extensions/mssql/src/sqlProjects/sqlProjectsService.ts [hunk 2]
```diff
@@ -32,9 +32,195 @@ export class SqlProjectsService implements mssql.ISqlProjectsService {
 		context.registerService(constants.SqlProjectsService, this);
 	}
 
+	public async addDacpacReference(projectUri: string, dacpacPath: string, suppressMissingDependencies: boolean, databaseVariable?: string, serverVariable?: string, databaseLiteral?: string): Promise<azdata.ResultStatus> {
+		const params: contracts.AddDacpacReferenceParams = { projectUri: projectUri, dacpacPath: dacpacPath, databaseVariable: databaseVariable, serverVariable: serverVariable, suppressMissingDependencies: suppressMissingDependencies, databaseLiteral: databaseLiteral };
+		try {
```
Comment (Charles-Gagnon) [inline extensions/mssql/src/sqlProjects/sqlProjectsService.ts:37]: Considering that you're using this pattern a whole bunch consider making a helper function for sending the request and handling errors.

File: extensions/types/vscode-mssql.d.ts [hunk 3]
```diff
@@ -429,6 +434,10 @@ declare module 'vscode-mssql' {
 		savePublishProfile(profilePath: string, databaseName: string, connectionString: string, sqlCommandVariableValues?: Record<string, string>, deploymentOptions?: DeploymentOptions): Thenable<ResultStatus>;
 	}
 
+	export interface ISqlProjectsService {
+		getDatabaseReferences(projectUri: string): Thenable<GetDatabaseReferencesResult>;
```
Comment (Charles-Gagnon) [inline extensions/types/vscode-mssql.d.ts:438]: Doc comments for new stuff

File: extensions/types/vscode-mssql.d.ts [hunk 4]
```diff
@@ -730,6 +739,66 @@ declare module 'vscode-mssql' {
 		deploymentOptions?: DeploymentOptions;
 	}
 
+	//#region ISqlProjectsService
+
+	//#region Parameters
+
+	export interface SqlProjectParams {
```
Comment (Charles-Gagnon) [inline extensions/types/vscode-mssql.d.ts:746]: Please update this in the VS Code MSSQL repo too.

**Review by Charles-Gagnon — APPROVED** [review]

File: extensions/mssql/src/mssql.d.ts (hunk 1 above)
Comment (Benjin) [reply extensions/mssql/src/mssql.d.ts:318]: I'm going to flesh out the conversion tool a little more to generate the C# docstrings, and will update this once I have that working.  Dunno if it'll make this PR.

Comment (Benjin) [reply extensions/mssql/src/mssql.d.ts:318]: They have made this PR!

File: extensions/types/vscode-mssql.d.ts (hunk 4 above)
Comment (Benjin) [reply extensions/types/vscode-mssql.d.ts:958]: Will do.

**Pull request merged**